### PR TITLE
Search planning areas

### PR DESCRIPTION
**Pull request by pflopez**

### CODE REVIEW

File: src/interface/src/app/standalone/planning-areas/planning-areas.datasource.ts [hunk 1]
```diff
@@ -8,21 +8,44 @@ import { QueryParamsService } from './query-params.service';
...
 
+  public sortOptions: Sort = this.queryParamsService.getInitialSortParams();
+  public pageOptions = this.queryParamsService.getInitialPageParams();
+  public searchTerm = this.queryParamsService.getInitialFilterParam();
+
+  /**
+   * Emits `true` if loading the first time or applying filters (where number of results change)
+   * `false` when done loading.
+   */
+  public initialLoad$ = this._initialLoad$.asObservable();
```
Comment (pflopez) [inline src/interface/src/app/standalone/planning-areas/planning-areas.datasource.ts:27]: @lastminutediorama what would you call this? Not sure `initialLoad` is clear enough

File: src/interface/src/app/standalone/planning-areas/planning-areas.datasource.ts [hunk 2]
```diff
@@ -8,21 +8,44 @@ import { QueryParamsService } from './query-params.service';
...
+    this._dataStream.asObservable(),
+  ]).pipe(
+    map(([initialLoad, results]) => !initialLoad && results.length === 0)
+  );
 
-  public pageOptions = this.queryParamsService.getInitialPageParams();
+  /**
+   * Emits `true` if applying filters or searching
+   */
+  public hasFilters$ = this._hasFilters$.asObservable();
```
Comment (pflopez) [inline src/interface/src/app/standalone/planning-areas/planning-areas.datasource.ts:48]: @lastminutediorama this one also would be something like hasFiltersOrSearch .... any ideas on a better name?

File: src/interface/src/app/standalone/planning-areas/planning-areas.component.html [hunk 3]
```diff
@@ -118,13 +119,24 @@ <h3>Home</h3>
...
-    </div>
-    <div>
-      You can explore the mapping tool without creating a planning area.
-    </div>
+
+    <ng-container *ngIf="hasFilters$ | async; else noData">
+      <div class="empty-state__title">No Planning Areas Found</div>
+      <div>No planning areas match the search criteria provided.</div>
+      <div>
+        Please verify spelling, use alternative keywords, or broaden the search
```
Comment (lastminutediorama) [inline src/interface/src/app/standalone/planning-areas/planning-areas.component.html:127]: Is this hasFilters state on the Figma anywhere? I'm wondering if it ought to have the same icon, or maybe something else to signify that nothing was found.

File: src/interface/src/app/standalone/planning-areas-search/planning-areas-search.component.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  selector: 'app-planning-areas-search',
+  standalone: true,
+  imports: [SearchBarComponent],
+  templateUrl: './planning-areas-search.component.html',
+  styleUrl: './planning-areas-search.component.scss',
+})
+export class PlanningAreasSearchComponent {
+  readonly key = 'planningAreasSearchHistory';
+
+  historyItems: string[] = this.getItems();
```
Comment (lastminutediorama) [inline src/interface/src/app/standalone/planning-areas-search/planning-areas-search.component.ts:14]: is it better to have this here or should this be in an ngInit?

File: src/interface/src/app/standalone/planning-areas/planning-areas.component.html (hunk 3 above)
Comment (pflopez) [reply src/interface/src/app/standalone/planning-areas/planning-areas.component.html:127]: its not but I pinged Julie and Jordan on the story to confirm

File: src/interface/src/app/standalone/planning-areas/planning-areas.datasource.ts (hunk 1 above)
Comment (lastminutediorama) [reply src/interface/src/app/standalone/planning-areas/planning-areas.datasource.ts:27]: It is a little confusing, but I can't think of another alternative. We can keep it for now, I think?

**Review by lastminutediorama — APPROVED** [review]
lgtm.

**Pull request merged**